tools/response_file.py

```diff
@@ -40,8 +40,20 @@ def escape(arg):
     if ' ' in arg:
       arg = '"%s"' % arg
     contents += arg + '\n'
-  with os.fdopen(response_fd, 'w') as f:
+
+  # When writing windows repsonse files force the encoding to UTF8 which we know
+  # that llvm tools understand.  Without this we will default whatever the local
+  # codepage might be which might be.
+  # See: https://github.com/llvm/llvm-project/blob/3f3d1c901d7abcc5b91468335679b1b27d8a02dd/llvm/include/llvm/Support/Program.h#L168-L170
+  # And: https://github.com/llvm/llvm-project/blob/63d16d06f5b8f71382033b5ea4aa668f8150817a/clang/include/clang/Driver/Job.h#L58-L69
+  if WINDOWS:
```

Comment:
Is there a reason not to use utf-8 on non-windows as well? Your first link here says `For UNIX, we always use UTF-8.` The second link says the encoding is ignored on non-windows, so I guess that's the danger?

Reply:
You might be right that we should maybe just always use UTF-8.  We haven't seen any issues on unix though.  I'll add a note here.